### PR TITLE
feat(daemon): Add Log Event harvest functionality

### PR DESCRIPTION
**Pull request by bduranleau-nr**
This PR implements the initial functionality required to harvest log events.
This will also allow integration_runner to handle EXPECT_LOG_EVENTS *.php integration tests

### CODE REVIEW

**Review by ZNeumann — APPROVED** [review]

**Review by mfulb — APPROVED** [review]
Thank you Brian!

**Comment by bduranleau-nr** [comment]
ok jenkins

File: src/newrelic/app_test.go [hunk 1]
```diff
@@ -375,7 +375,7 @@ func TestConnectPayloadEncoded(t *testing.T) {
 		`"metadata":{"NEW_RELIC_METADATA_ONE":"one","NEW_RELIC_METADATA_TWO":"two"},` +
 		`"identifier":"one;two",` +
 		`"utilization":{"metadata_version":1,"logical_processors":22,"total_ram_mib":1000,"hostname":"some_host"},` +
-		`"event_harvest_config":{"report_period_ms":60000,"harvest_limits":{"error_event_data":100,"analytic_event_data":10000,"custom_event_data":10000,"span_event_data":2323}}` +
+		`"event_harvest_config":{"report_period_ms":60000,"harvest_limits":{"error_event_data":100,"analytic_event_data":10000,"custom_event_data":10000,"span_event_data":2323,"log_event_data":10000}}` +
```
Comment (mfulb) [inline src/newrelic/app_test.go:378]: I have this in my current PR - I think we need to set the `info.AgentEventLimits.SpanEventConfig.Limit` to a specific value and check for that value in the generated JSON.
Perhaps we need another test to verify the defaults if agent limits are not supplied?

File: src/newrelic/app_test.go [hunk 2]
```diff
@@ -488,7 +488,7 @@ func TestConnectPayloadEncoded(t *testing.T) {
 		`"metadata":{},` +
 		`"identifier":"one;two",` +
 		`"utilization":{"metadata_version":1,"logical_processors":22,"total_ram_mib":1000,"hostname":"some_host"},` +
-		`"event_harvest_config":{"report_period_ms":60000,"harvest_limits":{"error_event_data":100,"analytic_event_data":10000,"custom_event_data":10000,"span_event_data":1001}}` +
+		`"event_harvest_config":{"report_period_ms":60000,"harvest_limits":{"error_event_data":100,"analytic_event_data":10000,"custom_event_data":10000,"span_event_data":1001,"log_event_data":10000}}` +
```
Comment (mfulb) [inline src/newrelic/app_test.go:491]: Like above I think setting `info.AgentEventLimits.SpanEventConfig.Limit` and checking the JSON would be a good idea.

File: src/newrelic/app_test.go (hunk 1 above)
Comment (bduranleau-nr) [reply src/newrelic/app_test.go:378]: SpanEvent or LogEvent?

**Review by mfulb — APPROVED** [review]
Looks good with the changes we made to the integration tests and Transaction.go/protocol.fbs.

**Comment by bduranleau-nr** [comment]
ok jenkins

**Pull request merged**